docs/howto/add_pagination_links.md

```diff
@@ -67,34 +54,50 @@ ex:
 }
 ```
 
-ActiveModelSerializers pagination relies on a paginated collection with the methods `current_page`, `total_pages`, and `size`, such as are supported by both [Kaminari](https://github.com/amatsuda/kaminari) or [WillPaginate](https://github.com/mislav/will_paginate).
-
+ActiveModelSerializers pagination relies on paginated collections which define the methods `#current_page`, `#total_pages`, and `#size`.
+Such methods are supported by both [Kaminari](https://github.com/amatsuda/kaminari) or [WillPaginate](https://github.com/mislav/will_paginate),
+but you can also roll out your own paginated collection by defining these methods.
```

Comment:
Well, some part did not really make sense e.g.:

```diff
-Although the others adapters does not have this feature, it is possible to
-implement pagination links to `JSON` adapter. For more information about it,
-please see in our docs 
```

Adding this to the JSONAPI section does not provide any value. If a user is looking for JSONAPI, he will not care about this. If he's looking for JSON, he'll probably just scroll down directly.

Then I think adding an explicit example instead of a small code snippet brings more value and is easier to understand. Here we know that we should use the pagination within an Rails controller:
```diff
-#array
-@posts = [1,2,3].paginate(page: 3, per_page: 1)
-render json: @posts
-
-#active_record
-@posts = Post.page(3).per_page(1)
-render json: @posts
-
+class PostsController < ApplicationController
+  def index
+    posts = Post.page(params[:page]).per_page(params[:per_page])
+    render json: posts
+  end
+end
```
It provides an example ready to use. Then the `#array`/`#active_record` examples are `Kaminari`/`WillPaginate` specific and does not make to much sense it. If users want to know how to use these tools, they should look the doc there.